flocker/control/_clusterstate.py

```diff
@@ -20,9 +23,32 @@ class ClusterStateService(Service):
 
     :ivar DeploymentState _deployment_state: The current known cluster
         state.
+    :ivar PMap _information_wipers: Map (wiper class, wiper key) to
+        (wiper, added timestamp), where wiper is a ``IClusterStateWipe``
+        provider.
+    :ivar _clock: ``IReactorTime`` provider.
     """
-    def __init__(self):
+    def __init__(self, reactor):
+        MultiService.__init__(self)
         self._deployment_state = DeploymentState()
+        timer = TimerService(1, self._wipe_expired)
+        timer.clock = reactor
+        timer.setServiceParent(self)
+        self._information_wipers = pmap()
+        self._clock = reactor
+
+    def _wipe_expired(self):
+        """
+        Clear any expired state from memory.
+        """
+        current_time = self._clock.seconds()
+        evolver = self._information_wipers.evolver()
+        for key, (wiper, timestamp) in self._information_wipers.items():
+            if current_time - 10 >= timestamp:
```

Comment:
Ordered sequence is problematic given you need to replace stuff by key when an update happens.